osquery/logger/plugins/http_logger.cpp

```diff
@@ -0,0 +1,195 @@
+  return Status(0, "OK");
+}
+
+class LogForwarder : public InternalRunnable{
+  LogForwarder(){}
+
+  ~LogForwarder(){}
+
+  /// Anything that needs to happen to verify that the LogForwarder can run
+  Status initVerify() {
+    if (FLAGS_max_data_size == 0) {
+      // VLOG(1) << "Could not start log forwarding loop, max data size not
+      // set";
+      return Status(1, "NOK");
+    }
+    if (FLAGS_log_force_send_time == 0) {
+      // VLOG(1) << "Could not start log forwarding loop, loop amount not set";
+      return Status(1, "NOK");
+    }
+    if (FLAGS_log_check_interval == 0) {
+      // VLOG(1) << "Could not start log forwarding loop, sleep amount not set";
+      return Status(1, "NOK");
+    }
+    if (FLAGS_log_force_send_threshold == 0) {
+      // VLOG(1) << "Could not start log forwarding loop, loop threshold not
+      // set";
+      return Status(1, "NOK");
+    }
+    if (!DBHandle::getInstance()->checkDB()) {
+      // VLOG(1) << "Could not start log forwarding loop, database checkDB
+      // failed";
+      return Status(1, "NOK");
+    }
+    return Status(0, "OK");
+  }
+
+  void enter() {
+    // Verify configuration is good
+    if (!initVerify().ok()) {
+      // We couldn't setup remote logging, so try to use filesystem
+      return;
+    }
+    std::vector<std::string> logs;
+
+    while (true) {
+      for (int i = 0; i < FLAGS_log_force_send_time / FLAGS_log_check_interval;
+           ++i) {
+        auto stat = DBHandle::getInstance()->Scan(kLogs, logs);
+        if (stat.ok()) {
+          // VLOG(1) << "LogForwarder encountered database error with message: "
+          //        << stat.getMessage();
+          return;
+        }
+        if (logs.size() > FLAGS_log_force_send_threshold) {
+          break;
+        }
+        logs.clear();
+        sleep(FLAGS_log_check_interval);
+      }
+
+      std::vector<std::string> send_blobs;
+      int bytes_of_data = FLAGS_max_data_size;
+      // Split our vector of returned strings into requests that fit the
+      // FLAGS_max_data_size
+      int last_blob_end = -1;
+      for (int i = 0; i < logs.size(); i++) {
+        bytes_of_data -= logs[i].length();
+        if (bytes_of_data < 0) {
+          //There are now too many bytes of data
+          if(FLAGS_max_data_size < logs[i].length()){
+            // This log is too big and cannot ever be sent, it must be dropped
+            bytes_of_data += logs[i].length();
+            // VLOG(1) << "Log: " << logs[i] << " is too big to be sent.";
+            logs.erase(logs.begin()+i);
+          }else{
+            std::string log_item;
+            for(int j=last_blob_end+1; j < i-1; j++){
+              log_item += (logs[j] + ", ");
+            }
+            log_item += (logs[i]);
+            last_blob_end = i;
+            send_blobs.push_back(log_item);
+          }
+        }
+      }
+      //Logs should be split into chunks of size small enough to send
+      for(const auto& item : send_blobs){
```

Comment:
This is stub code for later. In this loop, we will make and handle web requests.